File: ambeth/Ambeth.IoC/ambeth/ioc/hierarchy/SingletonContextHandle.cs
using De.Osthus.Ambeth.Log;

namespace De.Osthus.Ambeth.Ioc.Hierarchy
{
    public class SingletonContextHandle : AbstractSingletonContextHandle
    {
        [LogInstance]
        public new ILogger Log { private get; set; }

	    protected IServiceContext childContext;

	    protected override IServiceContext GetChildContext()
	    {
		    return childContext;
	    }

        protected override void SetChildContext(IServiceContext childContext)
	    {
		    this.childContext = childContext;
	    }
    }
}

File: jambeth/jAmbeth.IoC/src/main/java/de/osthus/ambeth/ioc/hierarchy/SingletonContextHandle.java
package de.osthus.ambeth.ioc.hierarchy;

import de.osthus.ambeth.ioc.IServiceContext;
import de.osthus.ambeth.log.ILogger;
import de.osthus.ambeth.log.LogInstance;

public class SingletonContextHandle extends AbstractChildContextHandle
{
	@SuppressWarnings("unused")
	@LogInstance
	private ILogger log;

	protected IServiceContext childContext;

	@Override
	protected IServiceContext getChildContext()
	{
		return childContext;
	}

	@Override
	protected void setChildContext(IServiceContext childContext)
	{
		this.childContext = childContext;
	}
}
